specification/src/chapter3.adoc

```diff
@@ -1,45 +1,312 @@
+
+Supervisor domains allow software components on the same hart to be developed, certified, deployed and attested independently of each other.
+
+A supervisor domain is a compartment above M-mode, physically isolated - memory, execution state, and devices - from other supervisor domains regardless of privilege level (below M-mode). Isolation and context switching between supervisor domains are managed by M-mode firmware. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NN
+| Either PMP/ePMP or MTT MUST be used to enforce physical memory isolation boundaries for supervisor domains, and to protect machine mode from any supervisor domain.
+
+|===
+
+PMP can be used for more static and deterministic use cases.
+
+MTT can be used where more fine grained dynamic resource management across supervisor domain boundaries is required.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+
+| ID#     
+| Requirement
+
+| CAT_NNN
+| A system supporting supervisor domains MUST support supervisor domain extensions for interrupts (Smsdia) and performance counters (TBD), and SHOULD support supervisor domain extensions for external debug (Smsdsd TBD).
+
+|===
+
+Interrupts: https://github.com/riscv/riscv-smmtt +
+External debug: https://github.com/riscv-non-isa/riscv-external-debug-security +
+Performance counters: 
+
+These extensions enable management of interrupts, external debug, and performance counters across supervisor domain boundaries. M-mode firmware should context switch hart HPM event/counters to manage isolation of performance counters:
+
+* External debug can be enabled for one supervisor domain without affecting other supervisor domains
+* M-mode firmware manage interrupt routing and preemption across supervisor domain boundaries
+* M-mode firmware can ensure that performance counters cannot be used by software in one supervisor domain to measure operations in other supervisor domains 
+
+==== MTT
+
+https://github.com/riscv/riscv-smmtt
+
+The _memory tracking table (MTT)_ is a memory structure managed by machine mode, tracking memory ownership across supervisor domains. It is designed to enable fine grained dynamic memory management across supervisor domain boundaries, with policy typically set by a hypervisor in a hosting domain responsible for resource management.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| Either PMP/ePMP or MTT MUST be used to protect M-mode from lower privilege levels
+
+| CAT_NNN  
+| MTT configurations MUST only be directly accessible to machine mode
+|===
+
+==== IOPMP
+
+https://github.com/riscv-non-isa/iopmp-spec
+
+IOPMP is a system level component providing physical memory access control for device initiated transactions, complementing PMP and sPMP rules. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| A system which supports PMP/ePMP, or sPMP, MUST implement IOPMP for device access control.
+
```

Comment:
Would it be better stated as : "A system MUST implement either MTT and/or IOPMP to control device initiated memory accesses"